lib/src/components/NxFilterInput/__tests__/NxFilterInput.test.tsx

```diff
@@ -5,160 +5,108 @@
+    expect(input).not.toHaveAttribute('validationErrors', 'It\'s all wrong');
+    expect(input).not.toHaveAttribute('type', 'textarea');
   });
 
-  it('puts the ref on the PrivateNxTextInput', function() {
+  it('sets ref on the Input', function() {
     const ref = React.createRef<HTMLDivElement>(),
+        el = renderEl({ ref } as Partial<Props>);
 
-        // note: the fragment is necessary to get around an enzyme issue:
-        // https://github.com/enzymejs/enzyme/issues/1852#issuecomment-433145879
-        component = mount(<><NxFilterInput { ...minimalProps } ref={ref} /></>),
-        domNode = component.find('div.nx-filter-input').getDOMNode();
-
-    expect(ref.current).toBe(domNode);
+    expect(ref.current).toBe(el);
   });
 
-  it('passes an faFilter icon with nx-icon--filter-icons as the prefixContent when searchIcon is undefined',
-      function() {
-        const IconFixture = function() {
-              return shallowComponent().prop('prefixContent');
-            },
-            icon = shallow(<IconFixture />);
-
-        expect(icon).toMatchSelector(NxFontAwesomeIcon);
-        expect(icon).toHaveProp(icon, faFilter);
-        expect(icon).toHaveClassName('nx-icon--filter-icons');
-      }
-  );
-
-  it('passes an faSearch icon with nx-icon--filter-icons as the prefixContent when searchIcon is true', function() {
-    const IconFixture = function() {
-          return shallowComponent({ searchIcon: true }).prop('prefixContent');
-        },
-        icon = shallow(<IconFixture />);
-
-    expect(icon).toMatchSelector(NxFontAwesomeIcon);
-    expect(icon).toHaveProp(icon, faSearch);
-    expect(icon).toHaveClassName('nx-icon--filter-icons');
+  it('renders a filter icon when searchIcon is undefined', function() {
+    const icons = quickRender().getAllByRole('img', { hidden: true });
+
+    expect(icons[0]).toHaveAttribute('data-icon', 'filter');
   });
 
-  it('passes an NxButton with the Close icon and nx-btn--clear classname as the suffixContent', function() {
-    const BtnFixture = function() {
-          return shallowComponent().prop('suffixContent');
-        },
-        btn = shallow(<BtnFixture />);
+  it('renders a search icon when searchIcon is true', function() {
+    const icons = quickRender({ searchIcon: true }).getAllByRole('img', { hidden: true });
 
-    expect(btn).toExist();
-    expect(btn).toMatchSelector(NxButton);
-    expect(btn.children()).toMatchSelector(Close);
-    expect(btn).toHaveClassName('nx-btn--clear');
+    expect(icons[0]).toHaveAttribute('data-icon', 'search');
   });
 
-  it('sets the title as "Clear filter" to the Nxbutton when searchIcon is undefined', function() {
-    const BtnFixture = function() {
-          return shallowComponent().prop('suffixContent');
-        },
-        btn = shallow(<BtnFixture />);
+  it('renders a button with an accessible name of "Clear filter" when searchIcon is undefined', async function() {
```

Comment:
As in the other PR, also test `false` and `null`.

Reply:
Updated here 85138bcbbccb76c45f2c9c54cc9b8a3ab70a7c4b.